src/popup/router/components/Modals/TransferReceive.vue

```diff
@@ -0,0 +1,294 @@
+<template>
+  <Modal
+    class="transfer-receive-modal"
+    has-close-button
+    from-bottom
+    :header="$t('modals.receive.title')"
+    @close="closeModal"
+  >
+    <div
+      class="transfer-receive"
+      data-cy="top-up-container"
+    >
+      <div class="account-info">
+        <Avatar
+          class="avatar"
+          :address="accounts[idx].address"
+          :name="accounts[idx].name"
+        />
+        <Truncate
+          v-if="accounts[idx].name"
+          :str="accounts[idx].name"
+        />
+        <span
+          v-else
+          data-cy="account-name"
+          class="account-name"
+        >
+          {{ $t('pages.account.heading') }} {{ accountIdx + 1 }}
+        </span>
+      </div>
+
+      <div class="address-info">
+        <QrcodeVue
+          :value="getQRdata(accounts[idx].address)"
+          size="112"
+          class="qrcode"
+        />
+
+        <a
+          class="address"
+          target="_blank"
+          :href="explorerUrl()"
+          :class="{ copied }"
+        >
+          <Scrollable>
+            {{ formatAddress(accounts[idx].address) }}
+          </Scrollable>
+        </a>
+      </div>
+
+      <div class="request-specific-amount">
+        <RequestAmount
+          v-model="amount"
+          own-validation
+          :validation="{ min_value_exclusive: 0 }"
+          :label="$t('modals.receive.requestAmount')"
+          @handleAssetSelection="handleAssetSelection"
+        />
+      </div>
```

Comment:
Do we need to wrap it into another div?

Reply:
Not that much but it gives the separation of concerns so I would prefer to leave it as it is.